### PR TITLE
TicTacTroll

### PR DESCRIPTION
**Pull request by YKH000**
Add TicTacTroll game (Exp2)

### CODE REVIEW

File: web/src/App.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import Home from "./Home";
 import OnlineGame from "./OnlineGame";
 
 import GameExp from "./GameExp";
+import Exp2 from "./Exp2";
```
[the remark this answers is not in the document]
Comment (YKH000) [reply web/src/App.js:6]: Done

File: web/src/App.js [hunk 2]
```diff
@@ -28,6 +29,9 @@ function App() {
             <Route path="/experiment">
               <GameExp />
             </Route>
+            <Route path="/exp2">
```
[the remark this answers is not in the document]
Comment (YKH000) [reply web/src/App.js:32]: Done

File: web/src/Exp2.css [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  line-height: 34px;
+  height: 70px;
+  margin-right: -1px;
+  margin-top: -1px;
+  padding: 0;
+  text-align: center;
+  width: 70px;
+}
+
+.square.expa0,
```
[the remark this answers is not in the document]
Comment (YKH000) [reply web/src/Exp2.css:36]: Done

File: web/src/Exp2.js [hunk 4]
```diff
@@ -0,0 +1,451 @@
...
+        this.setState({ backgroundColor: "#a9f2f5" });
+      }
+    }
+  };
+
+  render() {
+    return (
+      <button
+        style={{ backgroundColor: this.state.backgroundColor }}
+        className={"square expa" + this.props.id}
```
[the remark this answers is not in the document]
Comment (YKH000) [reply web/src/Exp2.js:47]: Done (and TicTacTroll.css square class renamed to expsquare accordingly)

File: web/src/Exp2.js [hunk 5]
```diff
@@ -0,0 +1,451 @@
...
+        onClick={() => {
+          if (canGo(this.props.id, this.props.stepNumber, this.props.squares)) {
+            this.props.onClick();
+            this.toggleBackgroundColor();
+          }
+        }} //{this.props.onClick}
+        onMouseEnter={this.mouseEnter}
```
[the remark this answers is not in the document]
Comment (YKH000) [reply web/src/Exp2.js]: Done

**Review by simonjhau — APPROVED** [review]
Approved

**Pull request merged**